diff -urN accel-ppp/accel-pppd/CMakeLists.txt accel-ppp/accel-pppd/CMakeLists.txt
--- accel-ppp/accel-pppd/CMakeLists.txt	2012-05-16 23:18:41.485555629 +0400
+++ accel-ppp/accel-pppd/CMakeLists.txt	2012-05-16 23:16:30.000000000 +0400
@@ -15,6 +15,11 @@
 
 INCLUDE_DIRECTORIES(include)
 
+CONFIGURE_FILE(
+    "${PROJECT_SOURCE_DIR}/cmake/gcc_debug_fix.sh.in"
+    "${PROJECT_BINARY_DIR}/gcc_debug_fix.sh"
+    @ONLY)
+
 IF (MEMDEBUG)
 	ADD_DEFINITIONS(-DMEMDEBUG)
 	IF (VALGRIND)
@@ -93,6 +98,8 @@
 set_property(TARGET accel-pppd PROPERTY INSTALL_RPATH_USE_LINK_PATH FALSE)
 set_property(TARGET accel-pppd PROPERTY INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/accel-ppp)
 
+set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${PROJECT_BINARY_DIR}/gcc_debug_fix.sh")
+
 INSTALL(TARGETS accel-pppd
 	RUNTIME DESTINATION sbin
 )

diff -urN accel-ppp/cmake/gcc_debug_fix.sh.in accel-ppp/cmake/gcc_debug_fix.sh.in
--- accel-ppp/cmake/gcc_debug_fix.sh.in	1970-01-01 03:00:00.000000000 +0300
+++ accel-ppp/cmake/gcc_debug_fix.sh.in	2012-05-16 23:15:25.000000000 +0400
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+PROJECT_BINARY_DIR="@PROJECT_BINARY_DIR@"
+PROJECT_SOURCE_DIR="@PROJECT_SOURCE_DIR@"
+
+# shell script invoked with the following arguments
+# $(CXX) $(CXX_DEFINES) $(CXX_FLAGS) -o OBJECT_FILE -c SOURCE_FILE
+
+# extract parameters
+SOURCE_FILE="${@: -1:1}"
+OBJECT_FILE="${@: -3:1}"
+COMPILER_AND_FLAGS=${@:1:$#-4}
+
+# make source file path relative to project source dir
+SOURCE_FILE_RELATIVE="${SOURCE_FILE:${#PROJECT_SOURCE_DIR} + 1}"
+
+# make object file path absolute
+OBJECT_FILE_ABSOLUTE="$(pwd)/$OBJECT_FILE"
+
+cd "$PROJECT_SOURCE_DIR"
+
+# invoke compiler
+exec $COMPILER_AND_FLAGS -c "${SOURCE_FILE_RELATIVE}" -o "${OBJECT_FILE_ABSOLUTE}"
