--- accel-ppp/accel-pppd/CMakeLists.txt	2012-05-28 22:41:24.823797460 +0400
+++ accel-ppp/accel-pppd/CMakeLists.txt	2012-05-28 22:41:40.315797371 +0400
@@ -88,11 +88,3 @@
 
 INSTALL(FILES accel-ppp.conf.5 DESTINATION share/man/man5)
 
-IF (NOT DEFINED CPACK_TYPE)
-	INSTALL(FILES accel-ppp.conf DESTINATION /etc RENAME accel-ppp.conf.dist)
-
-	INSTALL(CODE "EXECUTE_PROCESS(COMMAND mkdir -p /var/log/accel-ppp)")
-	INSTALL(CODE "EXECUTE_PROCESS(COMMAND mkdir -p /var/run/accel-ppp)")
-	INSTALL(CODE "EXECUTE_PROCESS(COMMAND 'echo 0 > /var/run/accel-ppp/seq')")
-ENDIF (NOT DEFINED CPACK_TYPE)
-
